app/Models/QueryBuilders/SearchFormQueryBuilder.php

```diff
@@ -0,0 +1,193 @@
+    {
+        $this->projectIds = $projectIds;
+        return $this;
+    }
+
+    /**
+     * @var string
+     */
+    private $negativeQuery;
+
+    public function negativeQuery(string $negativeQuery): self
+    {
+        $this->negativeQuery = $negativeQuery;
+        return $this;
+    }
+
+    public function buildQuery(): array
+    {
+        $queries = [];
+        $orQueries = [];
+        $andQueries = [];
+        $boolQueries = [];
+
+        if (isset($this->isPublic) && !empty($this->isPublic)) {
+            $andQueries[] = [
+                'term' => [
+                    'is_public' => $this->isPublic
+                ]
+            ];
+        }
+
+        if (isset($this->type) && !empty($this->type)) {
+            $andQueries[] = [
+                'term' => [
+                    'type' => $this->type
+                ]
+            ];
+        }
+
+        if (isset($this->subjectIds) && !empty($this->subjectIds)) {
+            $andQueries[] = [
+                'terms' => [
+                    'subject_id' => $this->subjectIds
+                ]
+            ];
+        }
+
+        if (isset($this->educationLevelIds) && !empty($this->educationLevelIds)) {
+            $andQueries[] = [
+                'terms' => [
+                    'education_level_id' => $this->educationLevelIds
+                ]
+            ];
+        }
+
+        if (isset($this->playlistIds) && !empty($this->playlistIds)) {
+            $andQueries[] = [
+                'terms' => [
+                    'playlist_id' => $this->playlistIds
+                ]
+            ];
+        }
+
+        if (isset($this->projectIds) && !empty($this->projectIds)) {
+            $andQueries[] = [
+                'terms' => [
+                    'project_id' => $this->projectIds
+                ]
+            ];
+        }
+
+        if (isset($this->query) && !empty($this->query)) {
+            $orQueries[] = [
+                'multi_match' => [
+                    'query' => $this->query,
+                    'fields' => ['title^5', 'name^5', 'description^3']
+                ]
+            ];
+
+            $orQueries[] = [
+                'multi_match' => [
+                    'query' => $this->query
+                ]
+            ];
+        }
+
+        if (isset($this->negativeQuery) && !empty($this->negativeQuery)) {
+            $boolQueries['must_not'] = [
+                'multi_match' => [
+                    'query' => $this->negativeQuery
+                ]
+            ];
+        }
+
+        if (!empty($andQueries)) {
+            $queries = $andQueries;
+        }
+
+        if (!empty($orQueries)) {
+            $queries[] = [
+                'bool' => [
+                    'should' => $orQueries
+                ]
+            ];
+        }
+
+        $boolQueries['must'] = $queries;
+
+        return [
+            'bool' => $boolQueries
+        ];
+    }
+}
```

Comment:
All PHP files MUST end with a non-blank line, terminated with a single LF.

Reply:
Added single linefeed at the end of SearchFormQueryBuilder.php.
Do let me know if you find it missing in any other of my files.